docs/content/basic/practices.rst

```diff
@@ -220,11 +228,29 @@ For example:
 Style
 -----
 
-Consistency make code more readable and easier for collaborators to jump in.
+Consistency makes code more readable and easier for collaborators to jump in.
 `PEP 8 <https://www.python.org/dev/peps/pep-0008/>`_ provides conventions for
 coding in Python. SimPEG is currently not `PEP 8
 <https://www.python.org/dev/peps/pep-0008/>`_ compliant, but we are working
-towards it and would appreciate contributions that do too! If you have sublime text 3, the linter can be set up through sublime text 3's package manager. The step by step process is found :ref:`here <api_practices_linter_install>`.
+towards it and would appreciate contributions that do too!
+
+SimPEG uses `black <https://black.readthedocs.io/>`_ version 23.1.0 to autoformat
+the code base, and all additions to the code are tested to ensure that they are
+compliant with `black`. We recommend installing `pre-commit <https://pre-commit.com/>`_
+hooks that are run on every commit to automatically ensure compliance.
+
+We also actively update the code base to ensure pep8 compliance by checking with
+`flake8 <https://flake8.pycqa.org/>`_ This performs style checks that could lead
+towards bugs, performs checks on consistent documentation formatting, or just
+identify poor coding practices. This is an ongoing process where we are fixing one
+style warning at a time. The fixed style warnings are checked to ensure no new code
+goes against an already established style. This test can also be installed locally
+using pre-commit hooks, similar to `black` above.
+
+
+If you have sublime text 3, the linter can be set up through sublime text 3's package manager. The step by
```

Comment:
I would put all these tips related to specific text editors inside their own subsection. They are related to the code style, but they are not part of the style rules that SimPEG follows. Replace Sublime Text with any other editor and the style check that SimPEG follows are the same.


As an alternative, what if we put all the content related to text editors in a separate page? Instead of including tips on how to configure them within the pages where the style rules and testing standards are laid out, move them to their own page. I think it has two benefits:

* It would be easier to update the text editor pages if some configuration gets outdated.
* It separates the "this is the standards we use to develop high quality code" and the "this is how you should configure your text editor". I think contributors might want to read these two topics in different moments (in one they learn, the other one is a how-to guide).

Reply:
At the very least they should be separated. However, I might be in favor of just removing these details all together. They are too specific for these areas to really be helpful, and why should we highlight one text editor over another?

Reply:
I also do not personally use sublime, so I actually have no clue if any of these recommendations are even relevant.

Reply:
I agree with your points. I don't think we should recommend one text editor over the rest. And personally, I wouldn't recommend one that is not open-source as the first option.

I think that we should remove the text editor sections, unless someone has a strong reason to keep them and is willing to keep them up-to-date.